bpf/flows.c

```diff
@@ -125,10 +136,25 @@ static inline int flow_monitor(struct __sk_buff *skb, u8 direction) {
     }
     flow_metrics *aggregate_flow = (flow_metrics *)bpf_map_lookup_elem(&aggregated_flows, &id);
     if (aggregate_flow != NULL) {
-        update_existing_flow(aggregate_flow, &pkt, len, filter_sampling);
+        if (aggregate_flow->if_index_first_seen == skb->ifindex) {
+            update_existing_flow(aggregate_flow, &pkt, len, filter_sampling);
+        } else {
+            // Only add info that we've seen this interface
+            additional_metrics *extra_metrics =
+                (additional_metrics *)bpf_map_lookup_elem(&additional_flow_metrics, &id);
+            if (extra_metrics != NULL) {
+                add_observed_intf(extra_metrics, skb->ifindex, direction);
+            } else {
+                additional_metrics new_metrics = {};
+                add_observed_intf(&new_metrics, skb->ifindex, direction);
+                bpf_map_update_elem(&additional_flow_metrics, &id, &new_metrics, BPF_NOEXIST);
```

Comment:
check and handle error cases and EEXISTS error

Reply:
done